src/calculation/CompositeReportBuilder.ts

```diff
@@ -0,0 +1,209 @@
+import { PopulationGroupResult, CalculationOptions, ExecutionResult } from '../types/Calculator';
+import { getCompositeScoringFromMeasure } from '../helpers/MeasureBundleHelpers';
+import { CompositeScoreType, PopulationType } from '../types/Enums';
+import { v4 as uuidv4 } from 'uuid';
+import { AbstractMeasureReportBuilder } from './AbstractMeasureReportBuilder';
+
+export type CompositeMeasureReport = fhir4.MeasureReport & {
+  group: [
+    fhir4.MeasureReportGroup & {
+      population: [
+        fhir4.MeasureReportGroupPopulation & { count: number },
+        fhir4.MeasureReportGroupPopulation & { count: number }
+      ];
+    }
+  ];
+};
+
+export class CompositeReportBuilder<T extends PopulationGroupResult> extends AbstractMeasureReportBuilder<
+  T,
+  CompositeMeasureReport
+> {
+  report: CompositeMeasureReport;
+  compositeScoringType: CompositeScoreType;
+  compositeFraction: { numerator: number; denominator: number };
+
+  constructor(compositeMeasure: fhir4.Measure, options: CalculationOptions) {
+    super(compositeMeasure, options);
+
+    this.compositeScoringType = getCompositeScoringFromMeasure(compositeMeasure) ?? 'all-or-nothing';
+    this.compositeFraction = { numerator: 0, denominator: 0 };
+    this.options = options;
+    this.report = {
+      resourceType: 'MeasureReport',
+      id: uuidv4(),
+      status: 'complete',
+      type: 'summary',
+      period: {
+        start: this.options.measurementPeriodStart,
+        end: this.options.measurementPeriodEnd
+      },
+      measure: compositeMeasure.url ?? 'UnknownMeasure',
+      group: [
+        {
+          population: [
+            {
+              count: 0,
+              code: {
+                coding: [
+                  {
+                    system: 'http://terminology.hl7.org/CodeSystem/measure-population',
+                    code: 'denominator'
+                  }
+                ]
+              }
+            },
+            {
+              count: 0,
+              code: {
+                coding: [
+                  {
+                    system: 'http://terminology.hl7.org/CodeSystem/measure-population',
+                    code: 'numerator'
+                  }
+                ]
+              }
+            }
+          ]
+        }
+      ]
+    };
+  }
+
+  public addPatientResults(results: ExecutionResult<T>) {
```

Comment:
```suggestion
  public addPatientResults(result: ExecutionResult<T>) {
```

More of a question and it isn't really important for this PR but I was thinking about it for my PR branched off of this branch... but thoughts on changing to be singular? It's being called in a loop of results so in `Calculator.ts` it is actually called as `addPatientResults(result)`. I know this was here before this PR but it actually may make things less confusing in the future but we could also deal with that later...

Reply:
https://github.com/projecttacoma/fqm-execution/pull/235/commits/d3657a34a5367c199725a566d599df445f8d025d